### DISCUSSION TITLE
Random white pixel on a slider

### DISCUSSION THREAD

**Discussion by uzervlad** [ppy/osu#21896] [Q&A] [answered]
Barely noticable but I figured I'd report it nonetheless.

Disappears as soon as the slider is hit. Doesn't appear with argon skin 

Map: https://osu.ppy.sh/beatmapsets/918900#osu/1918734

![osu_2022-12-28_21-13-11](https://user-images.githubusercontent.com/24654479/209855359-0b51dd4e-9de4-4791-8cab-fddde101f2dd.png)
![image](https://user-images.githubusercontent.com/24654479/209855601-b10739cd-d95b-4c3d-996e-65f4e099bb34.png)

**Comment by Paladynee** [comment]
how did you even notice that, might be a bug

**Comment by uzervlad** [comment]
Found another instance of this

Map: https://osu.ppy.sh/beatmapsets/742961#osu/1816113
07:18:595 (1)

Only appears with HR, doesn't disappear when the slider is hit

![image](https://user-images.githubusercontent.com/24654479/210157845-1f0bf2a1-4743-46c5-bb37-91ce6582800d.png)
![image](https://user-images.githubusercontent.com/24654479/210157882-2d6eb2b5-db35-496d-8c30-89b74bffe23c.png)

**Reply by peppy** [reply]
Can you confirm it always happens on the same slider? And also that it shows during watching autoplay?

**Reply by Joehuu** [reply]
Can confirm. Only seems to show at 1920x1080 fullscreen or borderless.

![osu_2022-12-31_20-45-32](https://user-images.githubusercontent.com/35318437/210161211-048bbf56-cb04-4029-9b7d-d89e68e06552.jpg)

**Reply by peppy** [reply]
Cool, the next step would be using the draw visualiser to figure if there's a stray drawable at that location.

**Reply by uzervlad** [reply]
Doesn't seem so, I couldn't find any.

Also when simulating HR using DA + MR it doesn't show up

**Reply by frenzibyte** [reply]
So far the only thing I've discovered is that the stray pixel is coming from the border colour (changing `ColourAt` to one solid colour hides the issue). My assumption is that one of the vertices of the path got positioned wrong given the specific vertices of that slider, but it's hard to tell with OpenGL on my platform (no RenderDoc-like app, only Metal debugging tools by Xcode).

**Reply by uzervlad** [reply]
I don't know much about how sliders are rendered, but it seems like a pixel from an "anchor" is not being drawn over with the slider body

<details>
<summary>RenderDoc captures</summary>

[Capture file](https://drive.google.com/file/d/1CgZxmYy6T55jKGpYghwsXeAhaK2CKmuB/view?usp=sharing)
![qrenderdoc_k3Vvk18J8W](https://user-images.githubusercontent.com/24654479/211192378-17d4e74a-e882-41ea-a714-0420ffafe5ee.gif)

[Capture file](https://drive.google.com/file/d/1rTBmStKK-gTlCE_6skdKxflqjTJszrDR/view?usp=sharing)
![qrenderdoc_tYGyt0VpLX](https://user-images.githubusercontent.com/24654479/211192454-ece84b16-937d-4c62-a143-3cbd22f44154.gif)
</details>

**Comment by Paladynee** [comment]
I found an instance of this on osu!stable too.
I was randomly playing a map then realized this pixel with nomod. Could recreate in +Auto too.
My storyboards + videos are disabled.

Mine however, does disappear after the slider is hit.

The white pixel seems to be unrelated to this setting: 
![image](https://user-images.githubusercontent.com/62181440/210706544-a8e99ade-f460-4562-9d77-38f4282d42aa.png)


Could recreate in 1440x900 (default monitor) fullscreen resolution.
Could recreate in windowed borderless 1440x900.
Could not recreate in 1280x720 fullscreen.
Could not recreate in 1280x800 fullscreen. (same aspect ratio with 1440x900 which the white pixel occurs at)
Could not recreate in windowed 800x600.
https://osu.ppy.sh/beatmapsets/1315340#osu/2725788
![screenshot173](https://user-images.githubusercontent.com/62181440/210705703-31d450ba-6f91-45f4-b96f-caa41d939724.png)

**Comment by peppy** [comment]
I think it's best to leave this until after we bring the new renderer online. It may fix it or may not.

**Comment by Glinte** [comment, marked as answer]
[osu-framework#5644](https://github.com/ppy/osu-framework/pull/5644) mentioned a probable cause of this issue.

**Reply by peppy** [reply]
It also contains a fix for it.
